test/sets.jl

```diff
@@ -216,6 +216,14 @@ u = unique([1,1,2])
 @test @inferred(unique(x for x in 1:1)) == [1]
 @test unique(x for x in Any[1,1.0])::Vector{Real} == [1]
 
+# unique!
+u = [1,1,2,3,1]
+unique!(u)
+@test length(u) == 3
+@test in(1,u)
+@test in(2,u)
+@test in(3,u)
+
```

Comment:
This just needs to test the actual value of `u`. That ensures that the length and order of occurrence is correct. In doing so I recommend swapping 3 and 2 in `u` to make it clearer. Perhaps something like
```julia
@testset "unique!" begin
    u = [1, 1, 3, 2, 1]
    unique!(u)
    @test u == [1, 3, 2]
end
```
Currently these tests don't test that the order of occurrence of the unique values is preserved.